### ISSUE TITLE
return statement(...) doesn't close session?

### ISSUE THREAD

**Issue by windmaple** [johnwheeler/flask-ask#70]
Hi,

I'm finding that return statement(...) doesn't close session? Here is my log:

*****New session started*****
*****Plane Strike launched*****
127.0.0.1 - - [06/Jan/2017 12:20:12] "POST / HTTP/1.1" 200 -
*****User provides strike position****
127.0.0.1 - - [06/Jan/2017 12:21:00] "POST / HTTP/1.1" 200 -
*****User confirms miss****
127.0.0.1 - - [06/Jan/2017 12:21:14] "POST / HTTP/1.1" 200 -
*****User stops****
127.0.0.1 - - [06/Jan/2017 12:21:20] "POST / HTTP/1.1" 200 -
*****New session started*****
ERROR:root:ERROR: Redundant Initialization. A session has already been started with sessionId: SessionId.4bf14310-b89a-4c6c-ae18-c003dd7fed31
*****Plane Strike launched*****

Basically I have the following code when user stops my skill:

```python
# stop game
@ask.intent('AMAZON.StopIntent')
def stop_game():
    print '*****User stops****'
    return statement('Thank you for playing Plane Strike!')
```

That error basically means it tries to initialize a tracking tool w/ the session object and it seems in this case the previous session object did not get closed out correctly when handling StopIntent. I also implemented @ask.session_ended but it seems it wasn't called.

Thoughts?

**Comment by johnwheeler** [comment]
it might be an intermittent problem with Alexa, but I haven't run across this problem. The code closes the session

https://github.com/johnwheeler/flask-ask/blob/master/flask_ask/models.py#L103

**Comment by windmaple** [comment]
I see. In this case, I can expect @ask.session_ended to be called, right?

**Comment by johnwheeler** [comment]
No. From the docs:

Your service receives a SessionEndedRequest when a currently open session is closed for one of the following reasons:

The user says “exit”.

The user does not respond or says something that does not match an intent defined in your voice interface while the device is listening for the user’s response.

An error occurs.

**Note that setting the shouldEndSession flag to true in your response also ends the session. In this case, your service does not receive a SessionEndedRequest.**

Your service cannot send back a response to a SessionEndedRequest.

**Comment by johnwheeler** [comment]
also note, "stop" is not equivalent to "exit" even though they seem similar. the former gives your skill a chance to respond and includes a built in intent (which you're using). the latter means to do a 'hard quit'. it's like crtl-alt-delete for skills.
